lib/Sapi.php

```diff
@@ -88,7 +88,15 @@ static function sendResponse(ResponseInterface $response) {
         if ($contentLength !== null) {
             $output = fopen('php://output', 'wb');
             if (is_resource($body) && get_resource_type($body) == 'stream') {
-                stream_copy_to_stream($body, $output, (int)$contentLength);
+                if (PHP_INT_SIZE !== 4){
+                    // use the dedicated function on 64 Bit systems
+                    stream_copy_to_stream($body, $output, $contentLength);  
+                } else {
+                    // workaround for 32 Bit systems to avoid stream_copy_to_stream
+                    while (!feof($body)) {
+                        fwrite($output, fread($body, 8192));    
```

Comment:
fwiw, 8192 seems low. I think you should push blocks 1MB at a time. It's not the 80's anymore ;)